x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -430,12 +434,113 @@ export function resultsServiceProvider(mlClusterClient: ILegacyScopedClusterClie
     return definition;
   }
 
+  async function getCategorizerStats(jobId: string, partitionByValue?: string) {
+    const mustMatchClauses: Array<Record<'match', Record<string, string>>> = [
+      {
+        match: {
+          result_type: 'categorizer_stats',
+        },
+      },
+    ];
+
+    if (typeof partitionByValue === 'string') {
+      mustMatchClauses.push({
+        match: {
+          partition_by_value: partitionByValue,
+        },
+      });
+    }
+    const results: SearchResponse<AnomalyCategorizerStatsDoc> = await callAsCurrentUser('search', {
```

Comment:
all searches to `ML_RESULTS_INDEX_PATTERN` should be performed by the internal user.